### PR TITLE
Give RVMO code path proper branch/version detection

### PR DESCRIPTION
**Pull request by mmazur**
Release branched builds can only work correctly if they're being run from a branch named `release-4.Y`.

This gets complicated in two places:
* If a developer wants to locally build from their feature branch, which definitively isn't named `release-4.Y`. This is taken care of by being able to run `RELEASE_BRANCH=release-4.16 make …` to override current branch detection.
* Appiface Jenkins just doesn't provide the build branch to the environment ([see](https://ci.int.devshift.net/view/osde2e/job/openshift-osd-example-operator-gh-build-release-4.16/74/injectedEnvVars/)) and the project's git repo is checked out as a detached HEAD commit, so there's no "what's my branch" information in there either. Best I can do right now is to try to extract the branch from JOB_NAME, which is what this code does… for now.

On the last point, there should be `GIT_LOCAL_BRANCH` available in the env, but for some reason it's not. I'm talking to appsre about this and if we ever get that working, I'll remove the current JOB_NAME hack and just look at `GIT_LOCAL_BRANCH`. But for now, this is the only way I can get this to work in prod.

### CODE REVIEW

**Comment by jbpratt** [comment]
/lgtm

needs approval from boilerplate approvers

**Comment by mmazur** [comment]
@2uasimojo @mjlshen hi, can I get a review?

**Review by mjlshen — APPROVED** [review]
/lgtm

**Pull request merged**
